Signal/src/ViewControllers/PhotoLibrary/PhotoLibrary.swift

```diff
@@ -0,0 +1,288 @@
+//
+//  Copyright (c) 2018 Open Whisper Systems. All rights reserved.
+//
+
+import Foundation
+import Photos
+import PromiseKit
+
+protocol PhotoLibraryDelegate: class {
+    func photoLibraryDidChange(_ photoLibrary: PhotoLibrary)
+}
+
+class PhotoMediaSize {
+    var thumbnailSize: CGSize
+
+    init() {
+        self.thumbnailSize = .zero
+    }
+
+    init(thumbnailSize: CGSize) {
+        self.thumbnailSize = thumbnailSize
+    }
+}
+
+class PhotoPickerAssetItem: PhotoGridItem {
```

Comment:
Much of these classes are unchanged - although I did some light renaming to generalize & be more consistent.  If you don't like the new names, I'd be happy to revise.

Reply:
Nah, the new names are better. :clap::clap::clap: